rust/agents/validator/src/submit.rs

```diff
@@ -146,15 +212,24 @@ impl ValidatorSubmitter {
     }
 }
 
+#[derive(Clone)]
 pub(crate) struct ValidatorSubmitterMetrics {
     latest_checkpoint_observed: IntGauge,
     latest_checkpoint_processed: IntGauge,
+    latest_legacy_checkpoint_observed: IntGauge,
+    latest_legacy_checkpoint_processed: IntGauge,
 }
 
 impl ValidatorSubmitterMetrics {
     pub fn new(metrics: &CoreMetrics, mailbox_chain: &HyperlaneDomain) -> Self {
         let chain_name = mailbox_chain.name();
         Self {
+            latest_legacy_checkpoint_observed: metrics
```

Comment:
Make sure you update the `latest_checkpoint()` docs as appropriate in `CoreMetrics`